src/Ilios/CoreBundle/Entity/Manager/PermissionManager.php

```diff
@@ -133,8 +133,8 @@ public function userHasReadPermissionToSchools(UserInterface $user, ArrayCollect
             'user'          => $user,
         ];
 
-        $schoolsWithReadPermission = $this->findPermissionBy($criteria);
-        $overlap = array_intersect($schools->toArray(), $schoolsWithReadPermission->toArray());
+        $schoolsWithReadPermission = $this->findPermissionsBy($criteria);
```

Comment:
good find. i will follow up with another pull request to fix the incorrect return value annotations for the various `findXsBy()` methods.